### DISCUSSION TITLE
Update messed up table name recognition

### DISCUSSION THREAD

**Discussion by dlogvin** [laravel/framework#33113] [General]
- Laravel Version: 7.14.0
- PHP Version: 7.2.5
- Database Driver & Version: mysql  Ver 14.14 Distrib 5.7.25

### Description:

Things first, first. My webapp worked GREAT before updating. 

The conflicting piece of code?
`$results = Subject::with(['user'])->where('user_id', $user_id)->withCount('homeworks');`

Why is laravel now asking for a table called `homework` when I'm literally trying to pull one called `homeworks`?

Base table or view not found: 1146 Table 'homewrk.homework'` (homewrk is the database name).

### Steps To Reproduce:

Create a table named in plural (idk, something like products), create a relationship to the user, and then inside the product, make another table called (again) something in plural, such as `characteristics`. Then try to show the products and the number of characteristics they have though the line of code I gave:

`$results = Product::with(['user'])->where('user_id', $user_id)->withCount('characteristics');`

It used to work, now it does not. I cannot even do `$subject->homeworks->count()` either.

**Comment by dlogvin** [comment]
adding `public $table = "homeworks";` to the model makes it work. But why I didn't have to put that before? it worked pretty well.

**Comment by rimace** [comment]
Actually, homework is uncountable and thus there is no "homeworks". Nevertheless if it was "fixed" to not produce a pluralized form it would be a breakage.
And why doesnt `$subject->homeworks->count()` work?
This completely depends on your implementation of the relation method. Can you post how you implemented `Subject::homeworks()`?

**Comment by driesvints** [comment]
Moved to discussions
